{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_read_1 = \"/media/eternus1/nfs/projects/nkureev/mini_reads/amp_res_1.fastq\" # 417842 ридов в сумме\n",
    "raw_read_2 = \"/media/eternus1/nfs/projects/nkureev/mini_reads/amp_res_2.fastq\"\n",
    "\n",
    "read_1_trimmomatic = \"/media/eternus1/nfs/projects/nkureev/mini_reads/trimmomatic/trimmomatic_done_pairedPE_1.fq\"\n",
    "read_2_trimmomatic = \"/media/eternus1/nfs/projects/nkureev/mini_reads/trimmomatic/trimmomatic_done_pairedPE_2.fq\"\n",
    "\n",
    "reads_1_v2trim = \"/media/eternus1/nfs/projects/nkureev/mini_reads/v2trim/minireads.trim_1.fastq\"\n",
    "reads_2_v2trim = \"/media/eternus1/nfs/projects/nkureev/mini_reads/v2trim/minireads.trim_2.fastq\"\n",
    "\n",
    "reads_1_fastp = \"/media/eternus1/nfs/projects/nkureev/mini_reads/fastp/fastp_done_1.fq\"\n",
    "reads_2_fastp = \"/media/eternus1/nfs/projects/nkureev/mini_reads/fastp/fastp_done_2.fq\"\n",
    "\n",
    "reads_1_trimgalore = \"/media/eternus1/nfs/projects/nkureev/mini_reads/trim_galore/trim_galore_done/amp_res_1_val_1.fq\"\n",
    "reads_2_trimgalore = \"/media/eternus1/nfs/projects/nkureev/mini_reads/trim_galore/trim_galore_done/amp_res_2_val_2.fq\"\n",
    "\n",
    "reads_1_prinseq = \"/media/eternus1/nfs/projects/nkureev/mini_reads/prinseq/prinseq_done.fq_good_out_R1.fastq\"\n",
    "reads_2_prinseq = \"/media/eternus1/nfs/projects/nkureev/mini_reads/prinseq/prinseq_done.fq_good_out_R2.fastq\"\n",
    "\n",
    "reads_1_adapterremoval = \"/media/eternus1/nfs/projects/nkureev/mini_reads/AdapterRemoval/AR_done_1.fq\"\n",
    "reads_2_adapterremoval =\"/media/eternus1/nfs/projects/nkureev/mini_reads/AdapterRemoval/AR_done_2.fq\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "class FastqObj(object):\n",
    "\n",
    "    def __init__(self, head, seq, strain, qual, phred33=False, forvard=True):\n",
    "        self.head = head.strip()\n",
    "        self.seq = seq.strip()\n",
    "        self.qual = qual.strip()\n",
    "        self.strain = strain.strip()\n",
    "        self.head_n = head_n.strip()\n",
    "        if forvard:\n",
    "            self.id = head.strip().replace(\"#\", \" \").split()[0].replace(\"@\",\"\") + \"_1\"\n",
    "        else:\n",
    "            self.id = head.strip().replace(\"#\", \" \").split()[0].replace(\"@\",\"\") + \"_2\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fastq_reader(fastq_file):\n",
    "    with open(fastq_file) as fh:\n",
    "        while True:\n",
    "            try:\n",
    "                head = fh.readline()\n",
    "                if head1 == '':\n",
    "                    break\n",
    "                seq = fh.readline()\n",
    "                strain = fh.readline()\n",
    "                qual = fh.readline()\n",
    "                fastq_obj = FastqObj(head, seq, strain, qual)\n",
    "                yield fastq_obj\n",
    "            except:\n",
    "                break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# returns table with 0 if read is the same and 1 if read was trimmed\n",
    "# columns is tools which was compared to base_tool\n",
    "\n",
    "def reads_comparer(base_tool_obj, compared_tools_list):\n",
    "    number_of_tools = len(compared_tools_list)\n",
    "    while True:\n",
    "        try:\n",
    "            reads_to_compare = defaultdict(list)\n",
    "            base_next = next(base_tool_obj)\n",
    "            for base_read in base_next:\n",
    "                base_id = base_read.id\n",
    "                base_seq = base_read.seq\n",
    "                reads_to_compare[base_id] = [0] * number_of_tools\n",
    "            n_tool = 0\n",
    "            for tool in compared_tools_list:\n",
    "                tool_next = next(tool)\n",
    "                for tool_read in tool_next:\n",
    "                    tool_id = tool_read.id\n",
    "                    tool_seq = tool_read.seq\n",
    "                    if tool_id == base_id and tool_seq != base_seq:\n",
    "                        reads_to_compare[base_id][n_tool] = 1\n",
    "                n_tool += 1\n",
    "            yield reads_to_compare\n",
    "        except:\n",
    "            break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# returns generator with read ID and trimmed sequence\n",
    "\n",
    "def reads_diff_finder(main_fastq_file, fastq_file_to_compare):\n",
    "    \n",
    "    def read_fastq(fastq_open):\n",
    "        head = fastq_open.readline().strip()\n",
    "        seq = fastq_open.readline().strip()\n",
    "        strain = fastq_open.readline().strip()\n",
    "        qual = fastq_open.readline().strip()\n",
    "        head_n = int(head[1:].split()[0].split(\".\")[1])\n",
    "        return head, seq, strain, qual, head_n\n",
    "    \n",
    "    \n",
    "    def reads_number(fastq_file):\n",
    "        reads_number = 0\n",
    "        with open(fastq_file) as fh:\n",
    "            for line in fh:\n",
    "                reads_number += 1\n",
    "        return int(reads_number/4)\n",
    "    \n",
    "    \n",
    "    main_file_len = reads_number(main_fastq_file)\n",
    "\n",
    "    proscock = False\n",
    "\n",
    "    with open(main_fastq_file) as fh1:    \n",
    "        with open(fastq_file_to_compare) as fh2:\n",
    "            for read in range(main_file_len):\n",
    "                if not proscock:\n",
    "                    head1, seq1, strain1, qual1, head1_n = read_fastq(fh1) \n",
    "                    head2, seq2, strain2, qual2, head2_n = read_fastq(fh2) \n",
    "                    if head2_n > head1_n: #if read was filtered it won't looking forwrd\n",
    "                        proscock = True\n",
    "                        continue\n",
    "                    elif head2_n == head1_n and seq1 != seq2:\n",
    "                        diff =  [head2, seq1.replace(seq2, \"\")]\n",
    "                        yield diff\n",
    "                        continue\n",
    "                else:\n",
    "                    head1, seq1, strain1, qual1, head1_n = read_fastq(fh1) \n",
    "                    if head2_n > head1_n: #if read was filtered it won't looking forwrd\n",
    "                        proscock = True\n",
    "                        continue\n",
    "                    elif head2_n == head1_n:\n",
    "                        if seq1 != seq2:\n",
    "                            diff =  [head2, seq1.replace(seq2, \"\")]\n",
    "                            yield diff\n",
    "                            proscock = False\n",
    "                            continue\n",
    "                        else:\n",
    "                            proscock = False\n",
    "                            continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def reads_number(fastq_file):\n",
    "    with open(fastq_file) as fh:\n",
    "        reads_number = int(len(fh.readlines())/4)\n",
    "    fh.close()\n",
    "    return reads_number"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def write_diff_in_file(base_reads_file, compared_files, tools_list, file_to_write):\n",
    "    for i in range(len(compared_files)):\n",
    "        print(f\"file {i}\")\n",
    "        \n",
    "        if i == 0: \n",
    "            mode = \"w\"\n",
    "        else: \n",
    "            mode = \"a\"\n",
    "        \n",
    "        raw_to_tool = reads_diff_finder(base_reads_file, compared_files[i])\n",
    "        with open(file_to_write, mode) as fw:\n",
    "            for p in raw_to_tool:\n",
    "                n_reads += 1\n",
    "                if n_reads % 100 == 0:\n",
    "                    print(n_reads)\n",
    "                header = p[0][1:].split()[0] + f\"_{tools_list[i]}\"\n",
    "                sequence = p[1]\n",
    "                fw.write(f\">{header}\\n{sequence}\\n\")\n",
    "        fw.close()\n",
    "    return f\"Successfully written in {file_to_write}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### write all diffs in files\n",
    "forward_diffs_all_tools = \"/media/eternus1/nfs/projects/nkureev/mini_reads/all_tools_trimmed_1.fasta\"\n",
    "reverse_diffs_all_tools = \"/media/eternus1/nfs/projects/nkureev/mini_reads/all_tools_trimmed_2.fasta\"\n",
    "tools_names = [\"v2trim\", \"trimmomatic\", \"fastp\", \"trimgolore\", \"prinseq\", \"adapterremoval\"]\n",
    "tools_to_compare_1 = [reads_1_v2trim, read_1_trimmomatic, reads_1_fastp, reads_1_trimgalore, reads_1_prinseq, reads_1_adapterremoval]\n",
    "tools_to_compare_2 = [reads_2_v2trim, read_2_trimmomatic, reads_2_fastp, reads_2_trimgalore, reads_2_prinseq, reads_2_adapterremoval]\n",
    "\n",
    "# print(write_diff_in_file(raw_read_1, tools_to_compare_1, tools_names, forward_diffs_all_tools))\n",
    "# print(write_diff_in_file(raw_read_2, tools_to_compare_2, tools_names, reverse_diffs_all_tools))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "base_reads_file = raw_read_2\n",
    "compared_files = tools_to_compare_2\n",
    "file_to_write = reverse_diffs_all_tools\n",
    "tools_list = tools_names\n",
    "n_reads = 0\n",
    "\n",
    "for i in range(len(compared_files)):\n",
    "    print(f\"file {i}\")\n",
    "\n",
    "    if i == 0: \n",
    "        mode = \"w\"\n",
    "    else: \n",
    "        mode = \"a\"\n",
    "\n",
    "    raw_to_tool = reads_diff_finder(base_reads_file, compared_files[i])\n",
    "    with open(file_to_write, mode) as fw:\n",
    "        for read_header in raw_to_tool:\n",
    "            n_reads += 1\n",
    "            if n_reads % 10000 == 0:\n",
    "                print(n_reads)\n",
    "            header = read_header[0][1:].split()[0] + f\"_{tools_list[i]}\"\n",
    "            sequence = p[1]\n",
    "            fw.write(f\">{header}\\n{sequence}\\n\")\n",
    "    fw.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "write_diff_in_file(raw_read_1, tools_to_compare_1, tools_names, forward_diffs_all_tools)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_to_tool = reads_difffinder(base_reads_file, reads_difffinder)\n",
    "forward = \"1\"\n",
    "\n",
    "for i in raw_to_tool:\n",
    "    header = i[0][1:].split()[0] + \"_{forward}_{tool}\"\n",
    "    sequence = i[1]\n",
    "    print(f\">{header}\\n{sequence}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "main_fastq_file, fastq_file_to_compare = raw_read_1, read_1_trimmomatic\n",
    "\n",
    "main_file_len = reads_number(main_fastq_file)\n",
    "\n",
    "proscock = False\n",
    "\n",
    "with open(main_fastq_file) as fh1:    \n",
    "    with open(fastq_file_to_compare) as fh2:\n",
    "        for read in range(main_file_len):\n",
    "            if not proscock:\n",
    "                head1 = fh1.readline().strip()\n",
    "                seq1 = fh1.readline().strip()\n",
    "                strain1 = fh1.readline()\n",
    "                qual1 = fh1.readline()\n",
    "                head1_n = int(head1[1:].split()[0].split(\".\")[1])\n",
    "                #print(head1, seq1)\n",
    "\n",
    "                head2 = fh2.readline().strip()\n",
    "                seq2 = fh2.readline().strip()\n",
    "                strain2 = fh2.readline()\n",
    "                qual2 = fh2.readline()\n",
    "                head2_n = int(head2[1:].split()[0].split(\".\")[1])\n",
    "                #print(head2, seq2)\n",
    "                \n",
    "                if head2_n > head1_n: #if read was filtered it won't looking forwrd\n",
    "                    proscock = True\n",
    "                    continue\n",
    "                elif head2_n == head1_n and seq1 != seq2:\n",
    "                    diff =  [head2, seq1.replace(seq2, \"\")]\n",
    "                    yield diff\n",
    "                    continue\n",
    "            else:\n",
    "                head1 = fh1.readline().strip()\n",
    "                seq1 = fh1.readline().strip()\n",
    "                strain1 = fh1.readline()\n",
    "                qual1 = fh1.readline()\n",
    "                head1_n = int(head1[1:].split()[0].split(\".\")[1])\n",
    "                if head2_n > head1_n: #if read was filtered it won't looking forwrd\n",
    "                    proscock = True\n",
    "                    continue\n",
    "                elif head2_n == head1_n:\n",
    "                    if seq1 != seq2:\n",
    "                        diff =  [head2, seq1.replace(seq2, \"\")]\n",
    "                        yield diff\n",
    "                        proscock = False\n",
    "                        continue\n",
    "                    else:\n",
    "                        proscock = False\n",
    "                        continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_reads_gen_1, raw_reads_gen_2 = fastq_pe_reader(raw_read_1), fastq_pe_reader(raw_read_2)\n",
    "v2trim_gen_1, v2trim_gen_2 = fastq_pe_reader(reads_1_v2trim), fastq_pe_reader(reads_2_v2trim)\n",
    "trimmomatic_gen_1, trimmomatic_gen_2 = fastq_pe_reader(read_1_trimmomatic), fastq_pe_reader(read_2_trimmomatic)\n",
    "fastp_gen_1, fastp_gen_2 = fastq_pe_reader(reads_1_fastp),  fastq_pe_reader(reads_2_fastp)\n",
    "trimgalore_gen_1, trimgalore_gen_2 = fastq_pe_reader(reads_1_trimgalore), fastq_pe_reader(reads_2_trimgalore)\n",
    "prinseq_gen_1, prinseq_gen_2 = fastq_pe_reader(reads_1_prinseq), fastq_pe_reader(reads_2_prinseq)\n",
    "adapterremoval_gen_1, adapterremoval_gen_2 = fastq_pe_reader(reads_1_adapterremoval), fastq_pe_reader(reads_2_adapterremoval)\n",
    "\n",
    "tools_to_compare_1 = [v2trim_gen_1, trimmomatic_gen_1, fastp_gen_1, trimgalore_gen_1, prinseq_gen_1, adapterremoval_gen_1]\n",
    "tools_to_compare_2 = [v2trim_gen_2, trimmomatic_gen_2, fastp_gen_2, trimgalore_gen_2, prinseq_gen_2, adapterremoval_gen_2]\n",
    "tools_names = [\"v2trim\", \"trimmomatic\", \"fastp\", \"trimgolore\", \"prinseq\", \"adapterremoval\"]\n",
    "file_with_diffs_1 = \"/media/eternus1/nfs/projects/nkureev/mini_reads/trimmed_parts_all_tools_1.fasta\"\n",
    "file_with_diffs_2 = \"/media/eternus1/nfs/projects/nkureev/mini_reads/trimmed_parts_all_tools_2.fasta\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "compared_tools_list = []\n",
    "for tool in tools_to_compare_1:\n",
    "    tool_obj = (t for t in tool)\n",
    "    compared_tools_list.append(tool_obj)\n",
    "base_tool_obj = (t for t in raw_reads_gen_1)\n",
    "\n",
    "number_of_tools = len(compared_tools_list)\n",
    "while True:\n",
    "    reads_to_compare = defaultdict(list)\n",
    "    base_next = next(base_tool_obj)\n",
    "    for base_read in base_next:\n",
    "        base_id = base_read.id\n",
    "        base_seq = base_read.seq\n",
    "        print(base_id)\n",
    "        reads_to_compare[base_id] = [0] * number_of_tools\n",
    "    n_tool = 0\n",
    "    for tool in compared_tools_list:\n",
    "        tool_next = next(tool)\n",
    "        for tool_read in tool_next:\n",
    "            tool_id = tool_read.id\n",
    "            tool_seq = tool_read.seq\n",
    "            if tool_id == base_id and tool_seq != base_seq:\n",
    "                reads_to_compare[base_id][n_tool] = 1\n",
    "        n_tool += 1\n",
    "    print(reads_to_compare)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_tools_compared_to_raw = reads_comparer(raw_reads_gen_1, tools_to_compare_1)\n",
    "\n",
    "all_reads = 0\n",
    "i = 0\n",
    "for read in all_tools_compared_to_raw:\n",
    "    for read_id, changes in read.items():\n",
    "        all_reads += 1\n",
    "        v2trim = changes[0]\n",
    "        trimmomatic = changes[1]\n",
    "        fastp = changes[2]\n",
    "        if changes[0] == 1:\n",
    "            i += 1\n",
    "print(all_reads, i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# сделать фаста файлик\n",
    "# хедер - id_tool\n",
    "# сиквенс - (raw_read - tool_read)\n",
    "\n",
    "# добавить next() чтбы вызывать строчки таблицы all_tools_compared_to_raw по очереди\n",
    "\n",
    "raw_reads_gen = fastq_pe_reader(fastq_file1=raw_read_1, fastq_file2=raw_read_2)\n",
    "v2trim_gen = fastq_pe_reader(fastq_file1=reads_1_v2trim, fastq_file2=reads_2_v2trim)\n",
    "trimmomatic_gen = fastq_pe_reader(fastq_file1=read_1_trimmomatic, fastq_file2=read_2_trimmomatic)\n",
    "fastp_gen = fastq_pe_reader(fastq_file1=reads_1_fastp, fastq_file2=reads_2_fastp)\n",
    "trimgalore_gen = fastq_pe_reader(fastq_file1=reads_1_trimgalore, fastq_file2=reads_2_trimgalore)\n",
    "prinseq_gen = fastq_pe_reader(fastq_file1=reads_1_prinseq, fastq_file2=reads_2_prinseq)\n",
    "adapterremoval_gen = fastq_pe_reader(fastq_file1=reads_1_adapterremoval, fastq_file2=reads_2_adapterremoval)\n",
    "\n",
    "raw_read_obj = (r for r in raw_reads_gen)\n",
    "v2trim_read_obj = (r for r in v2trim_gen)\n",
    "trimmomatic_read_obj = (r for r in trimmomatic_gen)\n",
    "fastp_read_obj = (r for r in fastp_gen)\n",
    "trimgalore_read_obj = (r for r in trimgalore_gen)\n",
    "prinseq_read_obj = (r for r in prinseq_gen)\n",
    "adapterremoval_read_obj = (r for r in adapterremoval_gen)\n",
    "tools_to_compare = [v2trim_read_obj, trimmomatic_read_obj, fastp_read_obj, trimgalore_read_obj, prinseq_read_obj, adapterremoval_read_obj]\n",
    "\n",
    "all_tools_compared_to_raw = reads_comparer(raw_read_obj, tools_to_compare)\n",
    "\n",
    "fasta_trimmed_parts = \"/media/eternus1/nfs/projects/nkureev/mini_reads/trimmed_parts.fasta\"\n",
    "\n",
    "with open(fasta_trimmed_parts, \"w\") as fw:\n",
    "    for read in all_tools_compared_to_raw:\n",
    "        for read_id, changes in read.items():\n",
    "            v2trim = changes[0]\n",
    "            trimmomatic = changes[1]\n",
    "            fastp = changes[2]\n",
    "            trimgalore = changes[3]\n",
    "            prinseq = changes[4]\n",
    "            adapterremoval = changes[5]\n",
    "            tools = [v2trim, trimmomatic, fastp, trimgalore, prinseq, adapterremoval]\n",
    "            tool_names = [\"v2trim\", \"trimmomatic\", \"fastp\", \"trimgalore\", \"prinseq\", \"adapterremoval\"]\n",
    "            for i in range(len(tools)):\n",
    "                if tools[i] == 1:\n",
    "                    print(read_id)\n",
    "                    for tool_read_id in tools_to_compare[i]:\n",
    "                        for read in tool_read_id:\n",
    "                            if read.id == read_id:\n",
    "                                print(read.seq)\n",
    "\n",
    "\n",
    "# подумать про сравнение тулов между собой"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_reads_gen = fastq_pe_reader(fastq_file1=raw_read_1, fastq_file2=raw_read_2)\n",
    "v2trim_gen = fastq_pe_reader(fastq_file1=reads_1_v2trim, fastq_file2=reads_2_v2trim)\n",
    "tools_list = [v2trim_gen]\n",
    "\n",
    "one_compared_to_raw = reads_comparer(raw_reads_gen, tools_list)\n",
    "\n",
    "i, j = 0, 0\n",
    "for read in one_compared_to_raw:\n",
    "    for read_id, changes in read.items():\n",
    "        j += 1\n",
    "        if changes[0] == 1:\n",
    "            i += 1\n",
    "print(j, i, i / j * 100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "different_reads = []\n",
    "losted_reads = []\n",
    "i = 0\n",
    "for header, seq in trimmomatic_reads.items():\n",
    "    if header in v2trim_reads.keys():\n",
    "        if seq != v2trim_reads[header]:\n",
    "            different_reads.append(header)\n",
    "    else:\n",
    "        losted_reads.append(header)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(len(v2trim_reads[different_reads[0]]))\n",
    "print(len(trimmomatic_reads[different_reads[0]]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "i = 0\n",
    "for header, seq in v2trim_reads.items():\n",
    "    if len(seq) == 251:\n",
    "        i += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(losted_reads)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "nontrimmed_by_trimmomatic = []\n",
    "for key in different_reads:\n",
    "    v2_seq = v2trim_reads[key]\n",
    "    trim_seq = trimmomatic_reads[key]\n",
    "    if len(v2_seq) > len(trim_seq):\n",
    "        print(v2_seq.replace(trim_seq, \"\"))\n",
    "    elif len(trim_seq) > len(v2_seq):\n",
    "        nontrimmed_by_trimmomatic.append(trim_seq.replace(v2_seq, \"\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "j = 0\n",
    "for i in nontrimmed_by_trimmomatic:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fastq_file1 = raw_read_1\n",
    "fastq_file2 = raw_read_2\n",
    "\n",
    "with open(fastq_file1) as fh1:\n",
    "        with open(fastq_file2) as fh2:\n",
    "            while True:\n",
    "                head1 = fh1.readline()\n",
    "                if not head1: break\n",
    "                seq1 = fh1.readline()\n",
    "                strain1 = fh1.readline()\n",
    "                qual1 = fh1.readline()\n",
    "                fastq_obj1 = FastqObj(head1, seq1, strain1, qual1)\n",
    "\n",
    "                head2 = fh2.readline()\n",
    "                seq2 = fh2.readline()\n",
    "                strain2 = fh2.readline()\n",
    "                qual2 = fh2.readline()\n",
    "                fastq_obj2 = FastqObj(head2, seq2, strain2, qual2, forvard=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in fas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## OLD STUFFF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_last_table(reads_to_compare_table, reads_to_compare_dir, *compared_tools_reads_dirs):\n",
    "    n_column = 1\n",
    "    for tool in compared_tools_reads_dirs:\n",
    "        reads_to_compare_table = between_tools_compare(reads_to_compare_table, reads_to_compare_dir, tool, n_column)\n",
    "        n_column += 1\n",
    "    return reads_to_compare_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# raw_reads compare with trimmomatic and v2_trim reads 0 - not equal, 1 - only one read equal, 2 - both reads equal\n",
    "# raw_reads_compare[header] = [trimmomatic, v2trim]\n",
    "def between_tools_compare(tool_to_compare_table, tool_to_compare_reads, compared_tool_reads, column_number):  \n",
    "    \n",
    "    for header, seq in tool_to_compare_reads.items():\n",
    "\n",
    "        tc_forward = seq[0]\n",
    "        tc_reverse = seq[1]\n",
    "\n",
    "        if header in compared_tool_reads:\n",
    "            ct_forward = compared_tool_reads[header][0]\n",
    "            ct_reverse = compared_tool_reads[header][1]\n",
    "\n",
    "            if tc_forward == ct_forward :\n",
    "                if tc_reverse == ct_reverse:\n",
    "                    tool_to_compare_table[header][column_number - 1] = 2\n",
    "                else:\n",
    "                    tool_to_compare_table[header][column_number - 1] = 1\n",
    "            elif tc_reverse == ct_reverse:\n",
    "                tool_to_compare_table[header][column_number - 1] = 1\n",
    "    return tool_to_compare_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def reads_reader(reads_file_1, reads_file_2):    \n",
    "    reads = {}\n",
    "    header = None\n",
    "    with open(reads_file_1) as fh1:\n",
    "        while True:\n",
    "            header = fh1.readline().split(\" \")[0][1:]\n",
    "            if not header:\n",
    "                break\n",
    "            seq = fh1.readline()\n",
    "            strand = fh1.readline()\n",
    "            strand_seq = fh1.readline()\n",
    "            reads[header] = [seq]\n",
    "            \n",
    "    with open(reads_file_2) as fh2:\n",
    "        while True:\n",
    "            header = fh2.readline().split(\" \")[0][1:]\n",
    "            if not header:\n",
    "                break\n",
    "            seq = fh2.readline()\n",
    "            strand = fh2.readline()\n",
    "            strand_seq = fh2.readline()\n",
    "            reads[header].append(seq)\n",
    "    return reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_compare_table(reads_dict, n_columns):\n",
    "    reads_to_compare = {}\n",
    "    for key in raw_reads.keys():\n",
    "        reads_to_compare[key] = [0]*n_columns\n",
    "    return reads_to_compare"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_reads_to_compare_table = create_compare_table(raw_reads, 6)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rr_table = create_last_table(raw_reads_to_compare_table, raw_reads, trimmomatic_reads, v2trim_reads)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(10):\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def reads_reader(reads_file_1, reads_file_2 ):    \n",
    "    reads = {}\n",
    "    header = None\n",
    "    with open(reads_file_1) as fh1:\n",
    "        for i, line in enumerate(fh1):\n",
    "            line = line.strip()\n",
    "            if line.startswith(\"@SRR\"):\n",
    "                if header:\n",
    "                    reads[header] = seq\n",
    "                header = line.split()[0][1:]\n",
    "                seq = []\n",
    "                header_switcher = True\n",
    "            elif header_switcher:\n",
    "                seq.append(line)\n",
    "                header_switcher = False\n",
    "    if header:\n",
    "        reads[header] = seq\n",
    "        \n",
    "    with open(reads_file_2) as fh2:\n",
    "        for i, line in enumerate(fh2):\n",
    "            line = line.strip()\n",
    "            if line.startswith(\"@SRR\"):\n",
    "                if header:\n",
    "                    reads[header].append(seq)\n",
    "                header = line.split()[0][1:]\n",
    "                seq = \"\"\n",
    "                header_switcher = True\n",
    "            elif header_switcher:\n",
    "                seq += line\n",
    "                header_switcher = False\n",
    "    if header:\n",
    "        reads[header].append(seq)\n",
    "    return reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "\n",
    "raw_read_1 = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/*_1.fastq\") # 417842 ридов в сумме\n",
    "raw_read_2 = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/*_2.fastq\")\n",
    "\n",
    "v2trim_1_reads = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/v2trim/*.trim_1.fastq\")\n",
    "v2trim_2_reads = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/v2trim/*.trim_2.fastq\")\n",
    "\n",
    "trimmomatic_1_reads = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/trimmomatic/*pairedPE_1.fq\")\n",
    "trimmomatic_2_reads = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/trimmomatic/*pairedPE_2.fq\")\n",
    "\n",
    "fastp_1_reads = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/fastp/*_1.fq\")\n",
    "fastp_2_reads = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/fastp/*_2.fq\")\n",
    "\n",
    "trimgalore_1_reads = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/trim_galore/trim_galore*/*_1_val*.fq\")\n",
    "trimgalore_2_reads = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/trim_galore/trim_galore*/*_2_val*.fq\")\n",
    "\n",
    "prinseq_1_reads = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/prinseq/*good_out_R1.fastq\")\n",
    "prinseq_2_reads = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/prinseq/*good_out_R2.fastq\")\n",
    "\n",
    "adapterremoval_1_reads = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/AdapterRemoval/*_1.fq\")\n",
    "adapterremoval_2_reads = glob.glob(\"/media/eternus1/nfs/projects/nkureev/hiseq*/AdapterRemoval/*_2.fq\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os.path\n",
    "diffinder = \"/home/dzilov/scripts/reads_comparer/reads_diff_finder.py\"\n",
    "tools_list = [\"v2trim\", \"trimmomatic\", \"fastp\", \"trimgolore\", \"prinseq\", \"adapterremoval\"]\n",
    "sequencers_list = [\"hiseq_2000\", \"hiseq2500\", \"hiseq4000\"]\n",
    "tools_inputs_1 = raw_read_1\n",
    "tools_inputs_2 = raw_read_2\n",
    "tools_outputs_1 = [v2trim_1_reads, trimmomatic_1_reads, fastp_1_reads, trimgalore_1_reads, prinseq_1_reads, adapterremoval_1_reads]\n",
    "tools_outputs_2 = [v2trim_2_reads, trimmomatic_2_reads, fastp_2_reads, trimgalore_2_reads, prinseq_2_reads, adapterremoval_2_reads]\n",
    "\n",
    "for i in range(len(raw_read_1)):\n",
    "    n = 0\n",
    "    for tool in tools_outputs_1: \n",
    "        output_file = os.path.dirname(tool[i]) + f\"/{tools_list[n]}_trimmed_parts_1.fasta\"\n",
    "        print(f\"time {diffinder} -b {raw_read_1[i]} -o {output_file} -c {tool[i]} -t {tools_list[n]}\" )\n",
    "        n += 1\n",
    "\n",
    "for i in range(len(raw_read_2)):\n",
    "    n = 0\n",
    "    for tool in tools_outputs_2: \n",
    "        output_file = os.path.dirname(tool[i]) + f\"/{tools_list[n]}_trimmed_parts_2.fasta\"\n",
    "        print(f\"time {diffinder} -b {raw_read_2[i]} -o {output_file} -c {tool[i]} -t {tools_list[n]}\" )\n",
    "        n += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
